### ISSUE TITLE
[Slider] [a11y]: role="slider" missing aria-label

### ISSUE THREAD

**Issue by valZho** [carbon-design-system/carbon#11676]
### Package

carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

7.57.2 (confirmed in latest pull)

### React version

16.14.x

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Assistive technology

unknown

### Description

> Each form control must have an associated label

> Form control with "slider" role has no associated label

The `Slider` component is missing a label

### WCAG 2.1 Violation

unknown

### Reproduction/example

https://nbwiqvmnp--github--3000.local.webcontainer.io

### Steps to reproduce

use the Slider component

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
